asdf.py

```diff
@@ -5,26 +5,55 @@
 
 class Brew(dotbot.Plugin):
     _supported_directives = ["asdf"]
+    _known_plugins = []
+    _has_checked_known_plugins = False
 
     def __init__(self, context):
         super(Brew, self).__init__(context)
+
+    @property
+    def known_plugins(self):
+        # Would love to use @cached_property but keeping python 2.7 compatibility
+        if self._has_checked_known_plugins:
+            return self._known_plugins
+
         output = self._run_command(
             "asdf plugin-list-all",
             error_message="Failed to get known plugins",
             stdout=subprocess.PIPE,
         )
-        plugins = output.decode("utf-8")
-        self._known_plugins = plugins.split()[::2]
+        plugins = output.decode("utf-8") # type: ignore
+        for plugin in plugins.split()[::2]:
+            self._known_plugins.append(plugin)
+
+        self._has_checked_known_plugins = True
+        return self._known_plugins
+
+    @property
+    def asdf_location(self):
+        return getattr(self, "_asdf_location", None)
+
+    @asdf_location.setter
+    def asdf_location(self, location):
+        self._asdf_location = location
```

Comment:
`_asdf_location` is good enough. We don't need this porperty.

Reply:
True, but this is more readable